### ISSUE TITLE
sudo python setup.py install can't work

### ISSUE THREAD

**Issue by haoxuhao** [1adrianb/face-alignment#25]
hi, when I run 'sudo python setup.py install', it can't work. 
the error is:
error in face_alignment setup command: 'install_requires' must be a string or list of strings containing valid project/version requirement specifiers

Could you help me? Thanks a lot.

**Comment by 1adrianb** [comment]
Hi @haoxuhao 

What version of python and setup tools are you using?
Can you please try to upgrade the setup tools: ``pip install --upgrade setuptools``

**Comment by haoxuhao** [comment]
I am very excited to see your reply.
My python version is 2.7.13 and operation system is ubuntu16.04 .Following your guide, I fixed this problem.
But when i run 'import face_alignment ' in terminal, it crashed.
Error shows: 'Segmentation fault (core dumped)'

**Comment by 1adrianb** [comment]
Hi @haoxuhao I suspect that this is not the only outdated package.
The simpliest thing to adress this will be to install miniconda (https://conda.io/miniconda.html) and create a virtual environment. Please try this and let me know if it worked.

**Comment by oyosc** [comment]
I meet this again,maybe python2?

**Comment by 1adrianb** [comment]
@oyosc it should work with both python2.7 and python 3.5+.  Are you using it in cpu or gpu mode? Unfortuntelly only with this information its really hard to indentify the problem. If you can run it using strace I can have a quick look other the logs. Otherwise, there is also a docker image provided wich will setup a self-contained container for you, but again, if configured properly a virtual env via conda should work.

**Comment by oyosc** [comment]
I try this again,and is okay,but when run import face_alignment in python2 cmd, it occurs ImportError: libboost_python.so.1.65.1: cannot open shared object file: No such file or directory

**Comment by 1adrianb** [comment]
@oyosc  please make sure that you libboost_python insalled. On ubuntu you can run: ``sudo apt-get install libboost-all-dev ``

**Comment by oyosc** [comment]
in centos is?

**Comment by 1adrianb** [comment]
@oyosc The issue seems to be unrelated to the package itself and it looks like the dlib version you installed is linked against a newer version of boost-devel (1.53 vs 1.65). Please make sure you compile dlib against the correct version of libboost, please check this:
https://stackoverflow.com/questions/2037425/boost-python-failed-with-libboost-python-so-1-41-0-cannot-open-shared-object-fi

**Comment by oyosc** [comment]
thx,but when i run the command of python detect_landmarks_in_image.py, it occurs ImportError: No module named torch!

**Comment by 1adrianb** [comment]
@oyosc you didn't install pytorch. Please follow the instructions carefully since this step is listed on the readme page. Again, if you still experience issues there is a docker image availalbe that will setup everything for you.

**Comment by oyosc** [comment]
I sure I have installed pytorch before,let me try again

**Comment by oyosc** [comment]
when i install pytorch again, it shows that thank you for installing Anaconda2!,but when i type anaconda-navigator,it tells me QXcbConnection: Could not connect to display;and when i type python detect_landmarks_in_image.py, the error is same

**Comment by 1adrianb** [comment]
@oyosc if you are using it over ssh make sure you are using X forwarding (ex: ssh -X ) since it seams that the display cannot be found

**Comment by oyosc** [comment]
i try this, but unfortunately no effect

**Comment by 1adrianb** [comment]
are you running it using a iphyton kernel? If yes, this may help you: https://github.com/ipython/ipython/issues/10627

**Comment by oyosc** [comment]
i try all of methods,but not ok,maybe i should try ubuntu? the system now is centos7

**Comment by oyosc** [comment]
@1adrianb hi,I have solved all of errors,but when i type python detect_landmarks_in_image.py,it shows:
Traceback (most recent call last):
  File "detect_landmarks_in_image.py", line 11, in <module>
    preds = fa.get_landmarks(input)[-1]
  File "build/bdist.linux-x86_64/egg/face_alignment/api.py", line 179, in get_landmarks
  File "build/bdist.linux-x86_64/egg/face_alignment/utils.py", line 82, in crop
  File "build/bdist.linux-x86_64/egg/face_alignment/utils.py", line 74, in transform
AttributeError: 'module' object has no attribute 'matmul'

**Comment by 1adrianb** [comment]
@oyosc it looks like you have an outdated version of pytorch. matmul was added in version 0.2 (http://pytorch.org/docs/master/torch.html?highlight=matmul#torch.matmul)
But then again the readme does state to use at least pytorch 0.2, please read it more carefully.

**Comment by oyosc** [comment]
@1adrianb yes, you are right;everything is ok,thank you very mush.

**Comment by oyosc** [comment]
@1adrianb hi,Are you still here? I need your help!

**Comment by 1adrianb** [comment]
@oyosc have you sorted out the issues?
